Improve media import error handling

balbuf opened the pull request:
- Addresses #3721
- This has 3 related parts, each isolated in their own commit
- Does not (yet) include any tests

1. When a local file is not found, continue to the next file (if there is one) instead of breaking - this matches similar behavior of other errors that allow the command to keep going and try with the following images.
1. Checks the return value of `download_url()` to see if it is a `WP_Error` - this is another thing that bit me when that function returned an error and `wp-cli` didn't really make it clear why the command failed.
1. With the `--porcelain` flag, make sure we return a non-zero exit code if there are any errors.

balbuf commented:
Let me know if you want any code comments or if anything should be changed/removed from this PR.

danielbachhuber commented:
> Let me know if you want any code comments or if anything should be changed/removed from this PR.

Code looks reasonable. Can you add tests please?

danielbachhuber commented:
@balbuf Will you be able to write the tests in the next couple of days? I'm releasing v1.1.0 next week, and would like to fit this in.

balbuf commented:
Hey @danielbachhuber! Let me see if I can get to that tomorrow.

balbuf commented:
Hey @danielbachhuber I added functional tests (including one to check that multiple images are imported) and also fixed a variable name (so the tests already paid off...)

Let me know what you think.

danielbachhuber commented:
@balbuf It looks like one of the tests is failing?

balbuf commented:
@danielbachhuber that failing test should be fixed now - we shall see!

balbuf commented:
hmm...seems like unrelated tests failed now due to some connectivity issues with downloads.wordpress.org. Can the tests be run again? Do they need to be?

danielbachhuber commented:
> Can the tests be run again?

I've restarted them

> Do they need to be?

Yep. Green light means go, red light means stop :)

balbuf commented:
[Not always](https://github.com/wp-cli/wp-cli/pulls?utf8=%E2%9C%93&q=is%3Apr%20is%3Amerged%20status%3Afailure) ;)

danielbachhuber commented:
> Not always ;)

Yeah yeah yeah. Wise guy....

Thanks for your work on this!
